File: src/connect-k-java/ManualAI.java
import java.util.*;
import java.io.*;

public class ManualAI extends AI 
{
	public ManualAI(int col, int row, int k, int g) 
	{
		super(col, row, k, g);
	}

	public Move GetMove(Move move) 
	{
		Scanner reader = new Scanner(System.in);
		int c, r;
		System.out.print("col, row: ");
		c = reader.nextInt();
		r = reader.nextInt();
		Move res = new Move(c, r);
		return res;
	}

}

File: src/connect-k-python/ManualAI.py
from BoardClasses import Move

class ManualAI():
    def __init__(self,col,row,k,g):
        pass

    def get_move(self,move):
        while True:
            try:
                c,r = map(lambda x:int(x),input("{col} {row}:").split(' '))
            except KeyboardInterrupt:
                raise KeyboardInterrupt
            except:
                print('invalid move')
                continue
            else:
                break
        return Move(c,r)
